pkg/mountpoint/args.go

```diff
@@ -0,0 +1,119 @@
+package mountpoint
+
+import (
+	"strings"
+
+	"k8s.io/apimachinery/pkg/util/sets"
+)
+
+const (
+	ArgForeground      = "--foreground"
+	ArgReadOnly        = "--read-only"
+	ArgAllowOther      = "--allow-other"
+	ArgAllowRoot       = "--allow-root"
+	ArgRegion          = "--region"
+	ArgCache           = "--cache"
+	ArgUserAgentPrefix = "--user-agent-prefix"
+	ArgAWSMaxAttempts  = "--aws-max-attempts"
+)
+
+// An Args represents arguments to be passed to Mountpoint during mount.
+type Args struct {
+	args sets.Set[string]
```

Comment:
I feel a set is the wrong data structure here. Why aren't we using a dictionary/hashmap? Then we don't have to iterate over the entire data structure to see presence, and we don't have to do as much messing around with formatting

Reply:
Not sure if Mountpoint currently uses such an argument but - dict/map is not a correct data structure to represent command line arguments IMO as you could have multiple values for same option, and also order of arguments might matter.

Like this `find` usage:
```
$ find . -name  "*file*" -type f -o \
         -name "*folder*" -type d
```

But that's being said the current implementation does not preserve original ordering and not sure if we need this behavior.

I didn't want to change original parsing function as that might cause some breaking changes if we do unintentional changes, but I think I added enough test cases now, and we should have enough confidence to make more refactor.

I think ideal data structure for this is: `List[Tuple[Key, Optional[Value]]]`

Reply:
Either an ordered map or what you proposed sounds ideal to me as I don't think MP currently supports multiple arguments with the same name.

Reply:
Address this in a separate PR: https://github.com/awslabs/mountpoint-s3-csi-driver/pull/349